### PR TITLE
[branch-1] HBASE-26075 Replication is stuck due to zero length wal file in oldWA…

### PR DESCRIPTION
**Pull request by shahrs87**
…Ls directory.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 56s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 40s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 46s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 41s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  5s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 49s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 41s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m  6s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  3s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 41s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 30s |  hbase-server: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 49s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 28s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 41s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   3m  0s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 136m 41s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 39s |  The patch does not generate ASF License warnings.  |
|  |   | 178m 38s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.TestReplicationKillSlaveRS |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3467 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 2a4eee3cee31 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-3467/out/precommit/personality/provided.sh |
| git revision | branch-1 / 5ece9cef |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/1/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/1/testReport/ |
| Max. process+thread count | 4489 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by shahrs87** [comment]
Failing test succeeded locally
```
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ hbase-server ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.hadoop.hbase.replication.TestReplicationKillSlaveRS
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 36.35 s - in org.apache.hadoop.hbase.replication.TestReplicationKillSlaveRS
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

**Comment by shahrs87** [comment]
@bharathv  could you please review the patch ? Thank you !

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 58s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 39s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 46s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 43s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  5s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 40s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m  9s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  6s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 41s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 46s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 46s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 31s |  hbase-server: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 50s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 27s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 42s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   2m 55s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 136m 26s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 39s |  The patch does not generate ASF License warnings.  |
|  |   | 178m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3467 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 2cb649870283 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-3467/out/precommit/personality/provided.sh |
| git revision | branch-1 / 5ece9cef |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/2/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/2/testReport/ |
| Max. process+thread count | 4123 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 50s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 41s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 45s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 41s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 50s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 40s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m  7s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  4s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 41s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 46s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 46s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 31s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 50s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 28s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 41s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   2m 57s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 135m 55s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 177m 40s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3467 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 6c2d2dec10de 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-3467/out/precommit/personality/provided.sh |
| git revision | branch-1 / 5ece9cef |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/3/testReport/ |
| Max. process+thread count | 4096 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/3/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReaderThread.java
Comment (bharathv) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReaderThread.java:319]: Not able to comment in L333, but that should not be queue.peek()? Also mind including the ioe stack in LOG.warn() while you are there.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/replication/regionserver/TestWALEntryStream.java
Comment (bharathv) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/replication/regionserver/TestWALEntryStream.java:944]: assertEquals() is better (will include the log with failures)

**Review by bharathv — COMMENTED** [review]
Couple of nits, but otherwise lgtm.

**Comment by shahrs87** [comment]
@bharathv  Addressed your comments. Please review again.

**Review by bharathv — APPROVED** [review]
Thanks, lets wait for the CI run before merging.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   6m 44s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 16s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 42s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 45s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 45s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 12s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 49s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 43s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m 10s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  6s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 41s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 47s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 47s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 39s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m  4s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 45s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 33s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 42s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   3m  7s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 141m  2s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 39s |  The patch does not generate ASF License warnings.  |
|  |   | 190m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3467 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4a39aff0593c 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-3467/out/precommit/personality/provided.sh |
| git revision | branch-1 / b7fbfdd |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/4/testReport/ |
| Max. process+thread count | 4346 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/4/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by gjacoby126 — APPROVED** [review]
+1

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 56s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 40s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 46s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 42s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 50s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 42s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m  7s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  6s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 40s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 31s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 50s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 32s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 31s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 40s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   2m 54s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 135m 59s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 38s |  The patch does not generate ASF License warnings.  |
|  |   | 178m  6s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3467 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 31cbe2a2ac41 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-3467/out/precommit/personality/provided.sh |
| git revision | branch-1 / b7fbfdd |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/5/testReport/ |
| Max. process+thread count | 4105 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3467/5/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**
